app/models/katello/file.rb

```diff
@@ -0,0 +1,32 @@
+module Katello
+  class File < Katello::Model
+    include Concerns::PulpDatabaseUnit
+
+    CONTENT_TYPE = Pulp::File::CONTENT_TYPE
+
+    has_many :repositories, :through => :repository_files, :class_name => "Katello::Repository"
+    has_many :repository_files, :class_name => "Katello::RepositoryFile", :dependent => :destroy, :inverse_of => :file
+
+    scoped_search :on => :name, :complete_value => true
+    scoped_search :on => :checksum
+    scoped_search :in => :repositories, :on => :name, :rename => :repository, :complete_value => true
+
+    def self.default_sort
+      order(:name)
+    end
+
+    def self.repository_association_class
+      RepositoryFile
+    end
+
+    def update_from_json(json)
+      keys = %w(name checksum)
+      custom_json = json.clone.delete_if { |key, _value| !keys.include?(key) }
```

Comment:
Seems weird to be using `clone` and `delete_if` instead of just `reject`. Better yet, just use slice:

```ruby
custom_json = json.slice('name', 'checksum')
```